modules/RTC/RTCUtils.js

```diff
@@ -547,19 +547,29 @@ function pollForAvailableMediaDevices() {
  * @returns {void}
  */
 function sendDeviceListToAnalytics(deviceList) {
-    const devicesPropsArray
-        = deviceList.map(
-            ({ deviceId, groupId, kind, label }) => {
-                // Filter the props of the device object.
-                return {
-                    deviceId,
-                    groupId,
-                    kind,
-                    label
-                };
-            });
+    const audioInputDeviceCount
+        = deviceList.filter(d => d.kind === 'audioinput').length;
+    const audioOutputDeviceCount
+        = deviceList.filter(d => d.kind === 'audiooutput').length;
+    const videoInputDeviceCount
+        = deviceList.filter(d => d.kind === 'videoinput').length;
+    const videoOutputDeviceCount
+        = deviceList.filter(d => d.kind === 'videooutput').length;
+
+    deviceList.forEach(device => {
+        const eventProperties = {
+            audioInputDeviceCount,
+            audioOutputDeviceCount,
+            videoInputDeviceCount,
+            videoOutputDeviceCount,
+            deviceId: device.deviceId,
+            deviceGroupId: device.groupId,
+            deviceKind: device.kind,
+            deviceLabel: device.label
+        };
 
-    Statistics.analytics.sendEvent(DEVICE_LIST, { devices: devicesPropsArray });
+        Statistics.analytics.sendEvent(AVAILABLE_DEVICE, eventProperties);
```

Comment:
Is there a reason why this was changed so that each device gets pushed up? Could data like device counts not be calculated/queried easily from the data itself?

Reply:
The new backend client that we're going to use does not support nesting attributes, so we can not (in a nice way) send a list of devices in a single event. And calculating the number of devices based on a number of events could be tricky as we'd have to find a good way to select the set of events.